hs_core/page_processors.py

```diff
@@ -50,14 +51,29 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
 
     metadata_status = _get_metadata_status(content_model)
 
-    relevant_tools = []
-    for res_type in ToolResourceType.objects.all():
-        if str(content_model.content_model).lower() in str(res_type.tool_res_type).lower():
-            url = res_type.content_object.url_bases.first()
-            if url:
-                tl = {'title': res_type.content_object.title,
-                      'url': "{}{}{}".format(url.value, "/?res_id=", content_model.short_id)}
-                relevant_tools.append(tl)
+
+    relevant_tools = None
+    if not resource_edit: # view mode
+        relevant_tools = []
+        for res_type in SupportedResTypes.objects.all():
+            if str(content_model.content_model).lower() in str(res_type.get_supported_res_types_str()).lower():
+                url_obj = res_type.content_object.url_bases.first()
+                tool_res_obj = hydroshare.get_resource_by_shortkey(url_obj.resShortID)
+                tool_edit_mode = False
+                if tool_res_obj.raccess.public or user.username == 'admin' or \
```

Comment:
Instead of doing this check here, you should call hs_core.views.utils.authorize() if possible.  It is a very bad practice to duplicate authorization logic.